numba/targets/arrayobj.py

```diff
@@ -1115,6 +1117,26 @@ def array_reshape(context, builder, sig, args):
     res = ret._getvalue()
     return impl_ret_borrowed(context, builder, sig.return_type, res)
 
+@builtin
+@implement('array.reshape', types.Kind(types.Array), types.VarArg(types.Any))
+def array_reshape_vararg(context, builder, sig, args):
+    # types
+    aryty = sig.args[0]
+    dimtys = sig.args[1:]
+    # values
+    ary = args[0]
+    dims = args[1:]
+    # coerce all types to uintp
```

Comment:
I think it should cast to intp (-1 is a documented valid value although our implementation doesn't support it).